app/src/main/java/org/eyeseetea/malariacare/fragments/PlanActionFragment.java

```diff
@@ -330,58 +351,47 @@ public boolean onBackPressed() {
 
     private void initSpinner(RelativeLayout llLayout) {
         actionSpinner = (CustomSpinner) llLayout.findViewById(R.id.plan_action_spinner);
-
-        secondaryActionSpinner = (CustomSpinner) llLayout.findViewById(
-                R.id.plan_action_secondary_spinner);
-
         final View secondaryView = llLayout.findViewById(R.id.secondaryView);
         final View otherView = llLayout.findViewById(R.id.otherView);
 
-        final ArrayAdapter<CharSequence> secondaryAdapter = ArrayAdapter.createFromResource(
-                llLayout.getContext(),R.array.plan_action_dropdown_suboptions,
-                android.R.layout.simple_spinner_item);
-        secondaryActionSpinner.setAdapter(secondaryAdapter);
-
         ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(llLayout.getContext(),
                 R.array.plan_action_dropdown_options, android.R.layout.simple_spinner_item);
         actionSpinner.setAdapter(adapter);
         actionSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
             @Override
-            public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
+            public void onItemSelected(AdapterView<?> adapterView, View view, int position,
+                    long l) {
                 String[] options = getResources().getStringArray(
                         R.array.plan_action_dropdown_options);
                 String selectedItem = adapterView.getItemAtPosition(position).toString();
-
+                String lastItem = mObsActionPlan.getAction1();
                 if (selectedItem.equals(options[0])) {
                     mObsActionPlan.setAction1(null);
-                    mCustomActionOtherEditText.setText("");
-                    secondaryActionSpinner.setSelection(0);
-                }else{
+                } else {
                     mObsActionPlan.setAction1(selectedItem);
                 }
-                if(mObsActionPlan.getAction2()==null || !mObsActionPlan.getAction2().equals(selectedItem)) {
+                if (lastItem == null || lastItem.equals(selectedItem)) {
```

Comment:
after merge I do not know if it makes sense this conditions with an empty body, can you review it please?